hlapipeline/utils/astrometric_utils.py

```diff
@@ -431,45 +431,51 @@ def extract_sources(img, **pars):
         # Remove likely cosmic-rays based on central_moments classification
         bad_srcs = np.where(classify_sources(cat) == 0)[0]+1
         segm.remove_labels(bad_srcs) # CAUTION: May be time-consuming!!!
-    cat = source_properties(img, segm)
 
-    
+
     # convert segm to mask for daofind
     if centering_mode == 'starfind':
         src_table = None
         #daofind = IRAFStarFinder(fwhm=fwhm, threshold=5.*bkg.background_rms_median)
         print("Setting up DAOStarFinder with: \n    fwhm={}  threshold={}".format(fwhm, bkg_rms_mean))
         daofind = DAOStarFinder(fwhm=fwhm, threshold=bkg_rms_mean)
-        # Identify nbrightest/largest sources 
+        # Identify nbrightest/largest sources
         if nlargest is not None:
+            if nlargest > len(segm.labels):
+                nlargest = len(segm.labels)
             large_labels = np.flip(np.argsort(segm.areas)+1)[:nlargest]
         print("Looking for sources in {} segments".format(len(segm.labels)))
-        
+
         for label in segm.labels:
             if nlargest is not None and label not in large_labels:
```

Comment:
Is it true that nlargest cannot be None at this stage?  If so, there is no reason to check the variable.

Reply:
By default, this variable starts as None (see line 393).  If it starts as None, it will stay as None and we want to make sure the correct action is taken here.